receiver/apachesparkreceiver/client.go

```diff
@@ -0,0 +1,184 @@
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http" // client defines the basic HTTP client interface.
+	"strconv"
+
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/apachesparkreceiver/internal/models"
+)
+
+const (
+	metricsPath      = "/metrics/json"
+	applicationsPath = "/api/v1/applications"
+)
+
+type client interface {
+	Get(path string) ([]byte, error)
+	GetClusterStats() (*models.ClusterProperties, error)
+	GetApplications() (*models.Applications, error)
+	GetStageStats(appID string) (*models.Stages, error)
+	GetExecutorStats(appID string) (*models.Executors, error)
+	GetJobStats(appID string) (*models.Jobs, error)
+}
+
+var _ client = (*apacheSparkClient)(nil)
+
+type apacheSparkClient struct {
+	client *http.Client
+	cfg    *Config
+	logger *zap.Logger
+}
+
+// newApacheSparkClient creates a new client to make requests for the Apache Spark receiver.
+func newApacheSparkClient(cfg *Config, host component.Host, settings component.TelemetrySettings) (client, error) {
+	client, err := cfg.ToClient(host, settings)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create HTTP Client: %w", err)
+	}
+
+	return &apacheSparkClient{
+		client: client,
+		cfg:    cfg,
+		logger: settings.Logger,
+	}, nil
+}
+
+// Get issues an authorized Get requests to the specified url.
+func (c *apacheSparkClient) Get(path string) ([]byte, error) {
+	req, err := c.buildReq(path)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		if err = resp.Body.Close(); err != nil {
+			c.logger.Warn("failed to close response body", zap.Error(err))
+		}
+	}()
+
+	if resp.StatusCode != http.StatusOK {
+		if resp.StatusCode >= 400 {
+			c.logger.Error("apachespark", zap.Error(err), zap.String("status_code", strconv.Itoa(resp.StatusCode)))
+		}
+		return nil, fmt.Errorf("request GET %s failed - %q", req.URL.String(), resp.Status)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body %w", err)
+	}
+
+	return body, nil
+}
+
+func (c *apacheSparkClient) GetClusterStats() (*models.ClusterProperties, error) {
+	body, err := c.Get(metricsPath)
+	if err != nil {
+		return nil, err
+	}
+
+	var clusterStats *models.ClusterProperties
+	err = json.Unmarshal(body, &clusterStats)
+	if err != nil {
+		return nil, err
+	}
+
+	return clusterStats, nil
+}
+
+func (c *apacheSparkClient) GetApplications() (*models.Applications, error) {
+	body, err := c.Get(applicationsPath)
+	if err != nil {
+		return nil, err
+	}
+
+	var apps *models.Applications
+	err = json.Unmarshal(body, &apps)
+	if err != nil {
+		return nil, err
+	}
+
+	return apps, nil
+}
+
+func (c *apacheSparkClient) GetStageStats(appID string) (*models.Stages, error) {
+	stagePath := fmt.Sprintf("/api/v1/applications/%s/stages", appID)
```

Comment:
Could use the `applicationsPath` variable for the first 3 elements on the path., same for executor & job stats